packages/dd-trace/src/log_channels.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const dc = require('diagnostics_channel')
+
+const Level = {
+  Debug: 'debug',
+  Info: 'info',
+  Warn: 'warn',
+  Error: 'error'
+}
+
+const _defaultLevel = Level.Debug
+
+class LogChannel {
+  constructor (name, logLevel) {
+    this.channel = dc.channel(`dd-trace:log:${name}`)
+    this.logLevel = logLevel
+  }
+
+  publish (message) {
+    if (this.channel.hasSubscribers) {
+      return this.channel.publish(message)
+    }
+  }
+
+  subscribe (onMessage) {
+    this.channel.subscribe(onMessage)
+  }
+
+  unsubscribe (onMessage) {
+    if (this.channel.hasSubscribers) {
+      this.channel.unsubscribe(onMessage)
+    }
+  }
+}
+
+// based on: https://github.com/trentm/node-bunyan#levels
+const logChannels = {};
+[Level.Debug, Level.Info, Level.Warn, Level.Error]
+  .forEach((level, index) => {
+    logChannels[level] = new LogChannel(level, (index + 2) * 10)
```

Comment:
I don't like the "magic" formula to calculate the number. In my opinion, it was better in the old way, where it was harcoded: 
```
const _logLevels = {
  'debug': 20,
  'info': 30,
  'warn': 40,
  'error': 50
}
```